active-rfcs/0000-name-mangling.md

```diff
@@ -0,0 +1,56 @@
+# Name
+
+- Start Date: 2021-12-05
+- RFC PR: [#2](https://github.com/boa-lang/rfcs/pull/2)
+- Implementation PR: (leave this empty)
+<!-- When when linking to the implementation PR, it should be done in the format of boa-lang/<repo>#<PR number> -->
+
+## Summary
+
+Defines how name mangling will work.
+
+## Basic example
+
+A method of the class `foo` with the signature `method(data: int) -> string` in the module `bar.baz` would be mangled to `boad2m3barm3bazc3fooi6methodp1t3i32r5string`
```

Comment:
Why not something like `boa__bar_baz__foo_method__int__string`?

Reply:
Hmm, I was mostly basing it off rust's mangling, which uses RLE (see prior art), but this could work too. It may be a bit messy to unmangle them, however

Reply:
```
leader__module__class_functionName__paramType_paramType_...__returnType
```


That's more what I meant